frontend/js/src/common/UserSearch.tsx

```diff
@@ -14,97 +12,96 @@ export type UserSearchProps = {
   initialValue?: string;
 };
 
-export type UserSearchState = {
-  newUser: string;
-  userSearchResults: Array<SearchUser>;
-};
+export default function UserSearch(props: UserSearchProps) {
+  // Context
+  const { APIService } = React.useContext(GlobalAppContext);
 
-export default class UserSearch extends React.Component<
-  UserSearchProps,
-  UserSearchState
-> {
-  static contextType = GlobalAppContext;
-  declare context: React.ContextType<typeof GlobalAppContext>;
+  // Props
+  const { onSelectUser, placeholder, clearOnSelect, initialValue } = props;
 
-  // eslint-disable-next-line react/sort-comp
-  constructor(props: UserSearchProps) {
-    super(props);
-    this.state = {
-      newUser: props.initialValue ?? "",
-      userSearchResults: [],
-    };
-  }
+  // States
+  const [newUser, setNewUser] = React.useState(initialValue ?? "");
+  const [userSearchResults, setUserSearchResults] = React.useState<
+    Array<SearchUser>
+  >([]);
 
-  throttledSearchUsers = _throttle(async () => {
-    await this.searchUsers();
-  }, 300);
+  // Refs
+  const dropdownRef = DropdownRef();
 
-  searchUsers = async () => {
-    const { APIService } = this.context;
-    const { newUser } = this.state;
-    try {
-      const response = await APIService.searchUsers(newUser);
-      this.setState({
-        userSearchResults: response.users,
-      });
-    } catch (error) {
-      this.handleError(error);
-    }
-  };
+  const searchUsers = React.useCallback(
+    async (newUserName: string) => {
+      try {
+        const response = await APIService.searchUsers(newUserName);
+        setUserSearchResults(response.users);
+      } catch (error) {
+        toast.error(<ToastMsg title="Error" message={error.message} />, {
+          toastId: "error",
+        });
+      }
+    },
+    [APIService]
+  );
 
-  handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState(
-      {
-        newUser: event.target.value,
-      },
-      this.throttledSearchUsers
-    );
-  };
+  const throttledSearchUsers = React.useCallback(
+    (newUserName: string) => {
+      _throttle(async () => {
+        await searchUsers(newUserName);
+      }, 300)();
+    },
+    [searchUsers]
+  );
```

Comment:
As far as I can tell, this does not throttle anything currently, because you are recreating the throttled function every call I think.
Basically, throttle returns a function, so currently every time `throttledSearchUsers` is called you call `throttle` and executed the function it returns, meaning no throttling is happening. See the network requests for each letter I typed:
![image](https://github.com/metabrainz/listenbrainz-server/assets/6179856/a503b391-b728-4813-bf3a-2465be5d570e)

Instead, I suggest using `useMemo` to memoize the throttled function with arbitrary arguments, rather than `useCallback` which requires you to pass at least one argument and complicates everything.

```suggestion
  const throttledSearchUsers = React.useMemo(
    () => _throttle(searchUsers, 800),
    [searchUsers]
  );
```
* Note that I also increased the throttle wait time to 800ms, 300 was just too low

The throttled result, typing really fast:
![image](https://github.com/metabrainz/listenbrainz-server/assets/6179856/4bca32a9-8d16-403a-b82c-13cc8e304e4d)

Reply:
Yes, this does fix the throttling issue. Thanks for the fix.